protocol/0048-DSRI-data_source_internal.md

```diff
@@ -51,6 +53,32 @@ on: {
 
 ```
 
+### 1.2.2 Repeating
+
+A repeating time trigger will contain a collection of timestamps and frequency tuples.
+For each tuple:
+
+- as soon as the protocol registers a time which is equal to or higher than the specified `timestamp` an event is emitted,
+- each time an integer multiple of the time period specified by the`frequency` field from the `timestamp` elapses another event should get fired.
+
+The repeating internal time triggered oracles will be used by the [perpetual futures](protocol/0053-PERP-product_builtin_perpetual_future.md) product, hence it must be possible to set them up to model a schedule like: every day at 04:00, 12:00 and 20:00.
+
+```rust
+on: {
+	{
+		timestamp: '202112311T04:00:00'
+		frequency: '24h'
+	},
+	{
+		timestamp: '202112311T12:00:00'
+		frequency: '24h'
+	},
+	{
+		timestamp: '202112311T20:00:00'
+		frequency: '24h'
+	},
+}
+
```

Comment:
TODO: 
- Add ACs.
- Be explicit about "frequencies" we allow.

Reply:
Made the repeating internal time triggered oracle bit more general for now, we can make it more prescriptive after discussing it with engineering.